antiplag/views.py

```diff
@@ -124,3 +123,55 @@ def get(self, request, id):
         else:
             # unprocessed submission documents should 'not exist' for the user
             return Response(status=status.HTTP_404_NOT_FOUND)
+
+
+class DocumentDiff(APIView):
+    def get(self, request, first_id, second_id):
+        first_document = get_object_or_404(Document, pk=first_id)
+        second_document = None
+        intervals = []
+
+        if first_document.submission.status == Submission.SubmissionStatus.PROCESSED:
+
+            for doc in first_document.result.matched_docs:
+                if doc["elastic_id"] == second_id:
+                    second_document = doc
+
+            intervals = second_document["intervals"]
+
+            if not second_document:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+
+            matches = []
+            second_text = preprocess_text(second_document["text"],
```

Comment:
Should we preprocess the text here? I think this operation is super-expensive for performance -therefore we do it in celery.

Doing this on endpoint blocks server for other users.

Reply:
Cannot we find the matches it on raw text instead? When comparing , the text from elastic wasn't preprocessed either, was it?

Reply:
refactored code. intervals are now saved directly in greedy string tiling algorithm.